sulong/mx.sulong/mx_sulong_suite_constituents.py

```diff
@@ -572,7 +572,12 @@ class CMakeProject(AbstractSulongNativeProject):  # pylint: disable=too-many-anc
     def __init__(self, suite, name, deps, workingSets, subDir, results=None, output=None, **args):
         super(CMakeProject, self).__init__(suite, name, deps, workingSets, subDir, results=results, output=output, **args)
         cmake_config = args.pop('cmakeConfig', {})
-        self.cmake_config = lambda: ['-D{}={}'.format(k, mx_subst.path_substitutions.substitute(v).replace('{{}}', '$')) for k, v in sorted(cmake_config.items())]
+        self.cmake_config = lambda: [
+            '-D{}={}'.format(k,
+                             mx_subst.path_substitutions.substitute(v)
+                                .replace('{{dollarsign}}', '$')
```

Comment:
It seems a bit strange to add an additional substitution mechanism since there is already one. I know this PR is just following a trend of the local code but it might be the right time to fix it rather than extend it.
This seems to process much more than paths so it should probably use `mx_subst.string_substitutions` and new string substitutions should be registered for `$` (`<dollarsign>`) and musl (`<has_musl_libc>`?). You can look for `string_substitutions.register_no_arg` to find examples.

Reply:
We fixed this already. `"\\$ORIGIN"` should now just work in `suite.py` (the `\\` escapes on the python level, resulting in the string containing `\$`, which mx interprets as literal `$`).